src/lightning/data/streaming/dataset_optimizer.py

```diff
@@ -547,9 +510,27 @@ def __init__(
         )
         self.random_seed = random_seed
 
-    def run(self) -> None:
+    def run(self, optimizable_dataset: _OptimizableDataset) -> None:
         """The `DatasetChunker.run(...)` method is used to trigger the data processing from your dataset into
         chunks."""
+        if not isinstance(optimizable_dataset, _OptimizableDataset):
+            raise ValueError(
+                """The provided argument to the DatasetOptimizer.run(...) needs to have the following format:
+
+                Example:
+
+                    class YourDataset:
```

Comment:
Relying on protocols feels very different from the rest of Lightning.
Shouldn't we just rely on providing a class with clear hooks that the user needs to implement?

Reply:
We can once we stabilise the API. @carmocca was against having this on the DataModule even If I believe this provides an easy onboarding.

Ideally, I would like to avoid adding a new object the user has to learn about just for 2 hooks. This seems like an overkill.

Reply:
Yes, that's entirely fine. Let's get user feedbacks first.

Reply:
Resolving for now as it would be addressed promptly.